src/main/kotlin/com/expedia/graphql/schema/generator/SchemaGenerator.kt

```diff
@@ -140,7 +140,7 @@ internal class SchemaGenerator(
         }
 
         if (!abstract) {
-            val dataFetcher: DataFetcher<*> = KotlinDataFetcher(target, fn, args)
+            val dataFetcher: DataFetcher<*> = KotlinDataFetcher(target, fn, args, config.hooks.dataFetcherExecutionPredicate)
```

Comment:
i dont think this makes much sense -> you are setting the same validator for ALL queries/functions which will execute against ALL parameters. 

Given current signature of the `DataFetcherExecutionPredicate` it just executes `test(value: T)` without any context on what it should be checking, e.g. lets say I add a validation to check if numbers are positive -> you would not only have to make the validator "smart" to skip all non number fields but it would also apply that check against ALL number fields (or maybe just ints or longs). 

IMHO in order for this to be useful, users should be able to define those validators per query/function (or even better per input param). Otherwise I am unsure how much value those global validators would give you.

Reply:
The mechanic is generic but then in terms of implementation and usages it will be done at the parameter / type level

Reply:
What I meant is that in the current implementation it takes just a value to be validated. There is no notion of which query/function nor which parameter is being evaluated. It only checks if value is correct which implies global validation logic.

After thinking about it I am unsure if adding extra validation hooks (whether through config, directives or some annotations) will be simpler than just implementing validation logic to the target method/functions.

Reply:
A possible usage is actually via annotation using bean validation.
Where the argument passed to the data fetcher are annotated.

For example:

```
class ValidatedQuery : Query {
    fun typeWithValidation(arg: NeedValidation): Int = arg.greaterThan2

    fun argumentWithValidation(@GraphQLValidation arg: TypeWithPattern): String = arg.lowerCaseOnly
}

@GraphQLValidation
data class NeedValidation(
    @field:Min(2)
    val greaterThan2: Int
)


data class TypeWithPattern(
    @field:Pattern(regexp = "[a-z]*")
    val lowerCaseOnly: String
)
```

Note that in this example `@GraphQLValidation` is actually a directive marking the that the element must be validated.
The advantage of using a directive here is it can be monitored via instrumentation.

The predicate logic must be totally from the usage and so using Bean Validation or a more manual approach is up to the consumer of the library if they chose to trigger the predicate.

Reply:
in the current logic you are still just passing raw value to be evaluated by the validator so it doesn't know anything about any annotations at all

Reply:
current logic is still too limited -> validation doesnt know anything about the target function (and I'd assume different functions might require different validations)

Reply:
for the sake of completeness -> due to other changes in the PR above comments were already addressed